packages/terra-table/src/subcomponents/Cell.jsx

```diff
@@ -0,0 +1,259 @@
+import React, {
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import * as KeyCode from 'keycode-js';
+import classNames from 'classnames/bind';
+import FocusTrap from 'focus-trap-react';
+import { injectIntl } from 'react-intl';
+import PropTypes from 'prop-types';
+
+import ThemeContext from 'terra-theme-context';
+import VisuallyHiddenText from 'terra-visually-hidden-text';
+
+import ColumnContext from '../utils/ColumnContext';
+import GridContext, { GridConstants } from '../utils/GridContext';
+import styles from './Cell.module.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * @private
+   * The intl object containing translations. This is retrieved from the context automatically by injectIntl.
+   */
+  intl: PropTypes.shape({ formatMessage: PropTypes.func }).isRequired,
+
+  /**
+   * String identifier of the row in which the Cell will be rendered.
+   */
+  rowId: PropTypes.string.isRequired,
+
+  /**
+   * String identifier of the column in which the Cell will be rendered.
+   */
+  columnId: PropTypes.string.isRequired,
+
+  /**
+   * The cell's row position in the table. This is zero based.
+   */
+  rowIndex: PropTypes.number,
+
+  /**
+   * The cell's column position in the table. This is zero based.
+   */
+  columnIndex: PropTypes.number,
+
+  /**
+   * Content that will be rendered within the Cell.
+   */
+  children: PropTypes.node,
+
+  /**
+   *  Boolean indicating if cell contents are masked.
+   */
+  isMasked: PropTypes.bool,
+
+  /**
+   * Boolean value indicating whether or not the column header is selectable.
+   */
+  isSelectable: PropTypes.bool,
+
+  /**
+   * Boolean indicating whether the Cell is currently selected.
+   */
+  isSelected: PropTypes.bool,
+
+  /**
+   * String that labels the cell for accessibility.
+   */
+  ariaLabel: PropTypes.string,
+
+  /**
+   * Boolean indicating that the cell is a row header.
+   */
+  isRowHeader: PropTypes.bool,
+
+  /**
+   * Boolean indicating that the cell has been highlighted.
+   */
+  isHighlighted: PropTypes.bool,
+
+  /**
+   * Callback function that will be called when this cell is selected.
+   */
+  onCellSelect: PropTypes.func,
+
+  /**
+   * String that specifies the height of the cell. Any valid CSS value is accepted.
+   */
+  height: PropTypes.string,
+};
+
+const defaultProps = {
+  isRowHeader: false,
+  isSelected: false,
+  isSelectable: false,
+  isMasked: false,
+};
+
+function Cell(props) {
```

Comment:
So just wanting to make sure, this package will be where all these subcomponents will live instead of in data-grid? So we plan on removing them from data-grid when consuming Table? I have some in-flight changes to both Row and Cell as part of Flowsheet that, depending on the PR review, I would need to make sure don't get lost in the process of migrating these.

Reply:
In the future, yes these components will all live inside this Table package. If you finish first, then I'll be adding your changes into this PR. If I finish first with this then we'll have to make another PR to add in-fight work to this PR when it happens. But most of it should be copy-paste and not require significant rework.

Reply:
Disregard, I'm changing my code to no longer affect Cell or Row.